src/new/wallet.js

```diff
@@ -859,6 +860,124 @@ class HathorWallet extends EventEmitter {
     return utxoDetails;
   }
 
+  /**
+   * Generates all available utxos
+   *
+   * @typedef {Object} Utxo
+   * @property {string} txId
+   * @property {number} index
+   * @property {string} tokenId
+   * @property {string} address
+   * @property {string} value
+   * @property {number} authorities
+   * @property {number|null} timelock
+   * @property {number|null} heightlock
+   * @property {boolean} locked
+   * @property {string} addressPath
+   *
+   * @param {Object} [options] Utxo filtering options
+   * @param {string} [options.token='00'] - Search for UTXOs of this token UID.
+   * @param {string|null} [options.filter_address=null] - Address to filter the utxos.
+   *
+   * @generator
+   * @function getAllUtxos
+   * @yields {Utxo} all available utxos
+   */
+  * getAllUtxos(options = {}) {
+    storage.setStore(this.store);
+    const historyTransactions = this.getFullHistory();
+
+    const { token, filter_address } = Object.assign({
+      token: HATHOR_TOKEN_CONFIG.uid,
+      filter_address: null,
+    }, options);
+
+    const utxos = [];
+    for (const tx_id in historyTransactions) {
+      const tx = historyTransactions[tx_id];
+      if (tx.is_voided) {
+        continue;
+      }
+
+      for (const [index, txout] of tx.outputs.entries()) {
+        if (
+          (filter_address && filter_address !== txout.decoded.address)
+          || txout.token !== token
+          || (!this.isAddressMine(txout.decoded.address))
+        ) {
+          continue;
+        }
+
+        if (txout.spent_by === null) {
+          if (wallet.canUseUnspentTx(txout, tx.height)) {
+            const isAuthority = (TOKEN_AUTHORITY_MASK & txout.token_data) > 0;
```

Comment:
```suggestion
            const isAuthority = wallet.isAuthorityOutput(txout);
```
The rule for authority tokens is pretty simple, but maybe re-using an existing validator that is available at this scope would be a better practice - even with the overhead of internally calling two more functions ( `isAuthorityOutput` and `transaction.isTokenDataAuthority` ).

What do you think?